libs/designer/src/lib/ui/mcp/parameters/parameterfield.tsx

```diff
@@ -0,0 +1,233 @@
+    }),
+    userRadioText: intl.formatMessage({
+      id: 'gOWYv9',
+      defaultMessage: 'User',
+      description: 'Label for parameter to use user input type',
+    }),
+    inputLabelText: intl.formatMessage({
+      id: 'gwYMqA',
+      defaultMessage: 'Input',
+      description: 'Label for input field in parameter editor radio button section',
+    }),
+  };
+
+  const onHandleRemoveError = useCallback(() => {
+    removeParameterError(parameter.id as string);
+  }, [parameter.id, removeParameterError]);
+
+  const isLargeParameter = useMemo(() => {
+    const editor = parameter.editor?.toLowerCase();
+    return (
+      editor === constants.EDITOR.ARRAY ||
+      editor === constants.EDITOR.DICTIONARY ||
+      editor === constants.EDITOR.HTML ||
+      editor === constants.EDITOR.TABLE
+    );
+  }, [parameter.editor]);
+
+  const onParameterValueChange = useCallback(
+    (newState: ChangeState) => {
+      const { value: newValueSegments, viewModel } = newState;
+      const propertiesToUpdate = { value: newValueSegments, preservedValue: undefined } as Partial<ParameterInfo>;
+
+      if (viewModel !== undefined) {
+        propertiesToUpdate.editorViewModel = viewModel;
+      }
+
+      const nodeId = operationId;
+      const parameterId = parameter.id as string;
+
+      if (parameterHasValue(propertiesToUpdate as ParameterInfo)) {
+        onHandleRemoveError();
+      }
+
+      const updateParameterAndDependencyPayload: UpdateParameterAndDependenciesPayload = {
+        nodeId,
+        groupId,
+        parameterId,
+        properties: propertiesToUpdate,
+        isTrigger: false,
+        operationInfo,
+        connectionReference: reference,
+        nodeInputs,
+        dependencies,
+        updateTokenMetadata: false,
+      };
+
+      dispatch(updateParameterAndDependencies(updateParameterAndDependencyPayload));
+
+      const updateParametersPayload: UpdateParametersPayload = {
+        nodeId,
+        parameters: [
+          {
+            groupId,
+            parameterId,
+            propertiesToUpdate: { value: newValueSegments },
+          },
+        ],
+        isUserAction: true,
+      };
+      dispatch(updateNodeParameters(updateParametersPayload));
+
+      onParameterVisibilityUpdate();
+    },
+    [
+      dispatch,
+      operationId,
+      groupId,
+      parameter.id,
+      operationInfo,
+      reference,
+      nodeInputs,
+      dependencies,
+      onParameterVisibilityUpdate,
+      onHandleRemoveError,
+    ]
+  );
+
+  const handleSelectionChange: RadioGroupProps['onChange'] = (_, data) => {
+    onParameterInputTypeChange(parameter.id, data.value as McpParameterInputType);
+    // Reset parameter value when model is selected
+    if (data.value === 'model') {
+      onHandleRemoveError();
+      onParameterValueChange({
+        value: [
+          {
+            id: parameter?.value?.[0]?.id,
+            type: 'literal',
+            value: '',
+          },
```

Comment:
use the function from './utils/parameters/segment.ts' createLiteralValueSegment('')